app/main.py

```diff
@@ -1,3 +1,17 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Any, Callable
+
+
+def cache(func: Callable) -> Any:
+    dict_of_results = {}
+
+    def inner(*args: Any, **kwargs: Any) -> Any:
+        func_arguments = args
+        if func_arguments not in dict_of_results:
+            print("Calculating new result")
+            dict_of_results[func_arguments] = func(*args, **kwargs)
+            return dict_of_results[func_arguments]
+        else:
```

Comment:
There is no need in this `else` statement, code will not execute after `return` anyway.